src/wallet/wallet.h

```diff
@@ -205,21 +205,29 @@ class CAddressBookData
 {
 private:
     bool m_change{true};
+    bool m_used{false};
     std::string m_label;
+    std::map<std::string, std::string> m_receive_requests;
 public:
     std::string purpose;
 
     CAddressBookData() : purpose("unknown") {}
 
-    typedef std::map<std::string, std::string> StringMap;
-    StringMap destdata;
-
     bool IsChange() const { return m_change; }
+    bool IsUsed() const { return m_used; }
+    void SetUsed(bool used) { m_used = used;}
     const std::string& GetLabel() const { return m_label; }
     void SetLabel(const std::string& label) {
         m_change = false;
         m_label = label;
     }
+    const std::map<std::string, std::string>& GetReceiveRequests() const { return m_receive_requests; }
+    bool SetReceiveRequest(std::string id, std::string value)
+    {
+        if (value.empty()) return m_receive_requests.erase(id);
```

Comment:
I strongly dislike using the empty string to indicate deletion. I think it would be better to have a separate `RemoveReceiveRequest`.

Reply:
> I strongly dislike using the empty string to indicate deletion. I think it would be better to have a separate `RemoveReceiveRequest`.

You are right. This is shitty and I will add a separate method to avoid continuing this pattern.

Reply:
> re: [#27224 (comment)](https://github.com/bitcoin/bitcoin/pull/27224#discussion_r1137647274)

This is done now